Content.Server/Humanoid/Systems/RandomHumanoidSystem.cs

```diff
@@ -0,0 +1,75 @@
+using Content.Server.Humanoid.Components;
+using Content.Server.RandomMetadata;
+using Content.Shared.Humanoid.Prototypes;
+using Content.Shared.Preferences;
+using Robust.Shared.Map;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Serialization.Manager;
+
+namespace Content.Server.Humanoid.Systems;
+
+/// <summary>
+/// This handles...
+/// </summary>
+public sealed class RandomHumanoidSystem : EntitySystem
+{
+    [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+    [Dependency] private readonly IComponentFactory _compFactory = default!;
+    [Dependency] private readonly ISerializationManager _serialization = default!;
+
+    [Dependency] private readonly HumanoidSystem _humanoid = default!;
+
+    /// <inheritdoc/>
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<RandomHumanoidComponent, MapInitEvent>(OnMapInit,
+            after: new []{ typeof(RandomMetadataSystem) });
+    }
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        foreach (var comp in EntityQuery<RandomHumanoidComponent>())
+        {
+            Del(comp.Owner);
```

Comment:
Update functions that only do cleanup like this are, well... Is there something stopping you from deleting the spawner after it's used instead of checking every frame if there are spawners that still exist?

Reply:
causes an exception when it's deleted during this part of entity creation, making it impossible to actually get the randomized humanoid

Reply:
Did you try QueueDel?